packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/di/app_dependencies.dart

```diff
@@ -5,34 +5,78 @@
 // license that can be found in the LICENSE file or at
 // https://opensource.org/licenses/MIT.
 
-{{#analytics}}
+import 'package:dio/dio.dart';
 import 'package:firebase_analytics/firebase_analytics.dart';
-import 'package:firebase_analytics/observer.dart';{{/analytics}}
+import 'package:firebase_analytics/observer.dart';
 import 'package:flutter/widgets.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
 import 'package:provider/provider.dart';
 import 'package:provider/single_child_widget.dart';
 
+import '../common_use_cases/fetch_auth_token_use_case.dart';
+import '../common_use_cases/logout_use_case.dart';
+import '../data_sources/local/auth_token_data_source.dart';
+import '../data_sources/remote/interceptors/analytics_interceptor.dart';
+import '../data_sources/remote/interceptors/auth_interceptor.dart';
+import '../repositories/auth_repository.dart';
+
 class AppDependencies {
   AppDependencies._(this.context);
 
-  factory AppDependencies.of(BuildContext context) => _instance != null
-      ? _instance!
-      : _instance = AppDependencies._(context);
+  factory AppDependencies.of(BuildContext context) =>
+      _instance != null ? _instance! : _instance = AppDependencies._(context);
 
   static AppDependencies? _instance;
 
   final BuildContext context;
 
   /// List of all providers used throughout the app
   List<SingleChildWidget> get providers => [
-    ..._analytics,
-  ];
+        ..._analytics,
+        ..._authDataSource,
+        ..._authRepo,
+        ..._logoutUseCase,
+        ..._fetchAuthTokenUseCase,
+        ..._httpClients,
+      ];
 
-  List<Provider> get _analytics => [{{#analytics}}
-      Provider<FirebaseAnalytics>(create: (context) => FirebaseAnalytics()),
-      Provider<FirebaseAnalyticsObserver>(
-        create: (context) =>
+  List<Provider> get _analytics => [
+        Provider<FirebaseAnalytics>(create: (context) => FirebaseAnalytics()),
+        Provider<FirebaseAnalyticsObserver>(
+          create: (context) =>
               FirebaseAnalyticsObserver(analytics: context.read()),
-      ),
-    {{/analytics}}];
+        ),
+      ];
+
+  List<Provider> get _authDataSource => [
+        Provider<AuthTokenDataSource>(
+            create: (context) =>
+                AuthTokenDataSource(const FlutterSecureStorage())),
+      ];
+
+  List<Provider> get _authRepo => [
+        Provider<AuthRepository>(
+            create: (context) => AuthRepository(context.read())),
+      ];
+
+  List<Provider> get _logoutUseCase => [
```

Comment:
What's the point of having an array when we always provide a single object? Usually we use an array to group some things... such as list of use cases :)

Reply:
Moved both use cases in one method get _useCases.